context/src/main/java/io/grpc/Context.java

```diff
@@ -878,11 +880,27 @@ public String toString() {
    */
   public abstract static class Storage {
     /**
+     * @deprecated This is an old API that is no longer used.
+     */
+    @Deprecated
+    public void attach(Context toAttach) {
+      throw new UnsupportedOperationException("Deprecated. Do not call.");
+    }
+
+    /**
      * Implements {@link io.grpc.Context#attach}.
      *
      * @param toAttach the context to be attached
+     * @return A {@link Context} that should be passed back into {@link #detach(Context, Context)}
+     *        as the {@code toRestore} parameter.
      */
-    public abstract void attach(Context toAttach);
+    public Context doAttach(Context toAttach) {
```

Comment:
We should document that the return value can be `null`, and what `Context.attach()` will use in that case. I should have had the same comment on `current()` too.